src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -468,30 +483,100 @@ public void postUsersOutboxProcess(Map<String, String> okapiHeaders, Handler<Asy
+            return succeededFuture(PutUsersByUserIdResponse.respond404WithTextPlain(entity.getId()));
+          }
+          boolean isConsortiaFieldsUpdated = isConsortiumUserFieldsUpdated(entity, userFromStorage);
+
+          return usersService.updateUser(conn, entity)
+            .compose(user -> {
+              if (isConsortiaFieldsUpdated) {
+                return userOutboxService.saveUserOutboxLog(conn, user, UserEvent.Action.EDIT, okapiHeaders);
+              }
+              return Future.succeededFuture();
+            })
+            .map(aVoid -> PutUsersByUserIdResponse.respond204())
+            .map(Response.class::cast);
+        })
+      .onComplete(reply -> {
+
+        if (reply.cause() != null) {
+          handleUpdateUserFailures(entity, asyncResultHandler, reply);
+          return;
+        }
+
+        userOutboxService.processOutboxEventLogs(vertxContext.owner(), okapiHeaders);
+        asyncResultHandler.handle(reply);
+      }));
+  }
+
+  private void handleUpdateUserFailures(User user, Handler<AsyncResult<Response>> asyncResultHandler, AsyncResult<Response> reply) {
+    String errorMessage = reply.cause().getMessage();
+    if (isDuplicateUsernameError(errorMessage)) {
+      logger.error("User with this username {} already exists", user.getUsername());
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(
+            "User with this username already exists")));
+      return;
+    }
+
+    if (isDuplicateBarcodeError(errorMessage)) {
+      logger.error("This barcode {} has already been taken", user.getBarcode());
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(
+            "This barcode has already been taken")));
+      return;
+    }
+
+    if (reply.cause() instanceof PgException) {
+      String errorMsg = ((PgException) reply.cause()).getDetail();
+      logger.error(errorMsg);
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(errorMsg)));
+      return;
+    }
+
+    logger.error(errorMessage);
+    asyncResultHandler.handle(
+      succeededFuture(PutUsersByUserIdResponse
+        .respond400WithTextPlain(errorMessage))
+    );
+  }
+
+  private boolean isConsortiumUserFieldsUpdated(User updatedUser, User userFromStorage) {
+    if (ObjectUtils.notEqual(userFromStorage.getUsername(), updatedUser.getUsername())) {
+      logger.info("The username has been updated to {}", updatedUser.getUsername());
+      return true;
+    }
+
+    Personal oldPersonal = userFromStorage.getPersonal();
+    Personal newPersonal = updatedUser.getPersonal();
+
+    if (oldPersonal == null && newPersonal == null) {
+      logger.info("Personal fields have not been updated");
+      return false;
+    }
+
+    if ((oldPersonal != null & newPersonal == null) || oldPersonal == null) {
```

Comment:
Unfortunately, this is not suitable here, because we need to check whether personal has been updated or not. If the oldPersonal and the newPersonal are both null or nonNull, then if should not be executed.